Cdk pipeline modern api typescript (#295)
aws-samples/aws-cdk-intro-workshop: workshop/content/20-typescript/70-advanced-topics/200-pipelines/3000-new-pipeline.md

## Define an Empty Pipeline
Now we are ready to define the basics of the pipeline.

> We will be using new package here, so first `npm install @aws-cdk/pipelines`.

Reviewer: ```suggestion
We will be using a new package here, so first `npm install @aws-cdk/pipelines`.
```

            repositoryName: "WorkshopRepo"
        });

        // The basic pipeline declaration. This sets the initial structure
        // of our pipeline
       const pipeline = new CodePipeline(this, 'Pipeline', {
            pipelineName: 'WorkshopPipeline',
            synth: new CodeBuildStep('SynthStep', {
                    input: CodePipelineSource.codeCommit(repo, 'master'),
                    installCommands: [
                        'npm install -g aws-cdk'
                    ],
                    commands: [
                        'npm ci',
                        'npm run build',
                        'npx cdk synth'
                    ]
                }
            )
        });
    }
}
{{</highlight>}}

### Component Breakdown
The above code does several things:

* `new CodePipeline(...)`: This initializes the pipeline with the required values. This will serve as the base component moving forward. Every pipeline requires at bare minimum:

>     * `synth(...)`: The `synthAction` of the pipeline will take the source artifact generated in by the `input` and build the application based on the `commands`. This is always followed by `npx cdk synth`.

Reviewer: ```suggestion
    * `synth(...)`: The `synthAction` of the pipeline describes the commands necessary to install dependencies, build, and synth the CDK application from source. This should always end in a *synth* command, for NPM-based projects this is always `npx cdk synth`.
```

>   The `input` of the synth step will check the designated repository for source code and generate an artifact.

Reviewer: ```suggestion
  * The `input` of the synth step specifies the repository where the CDK source code is stored.
```

## Deploy Pipeline and See Result
All that's left to get our pipeline up and running is to commit our changes and run one last cdk deploy. 

```
git commit -am "MESSAGE" && git push
npx cdk deploy
```

> CodePipelines auto-update for each commit in a source repo, so this is the *last time* we will need to execute this command!

Reviewer: ```suggestion
CDK Pipelines auto-update for each commit in a source repo, so this is the *last time* we will need to execute this command!
```
